x-pack/legacy/plugins/fleet/dev_docs/fleet_agents_interactions_detailed.md

```diff
@@ -0,0 +1,46 @@
+# Fleet <-> Agent Interactions
+
+## Agent enrollment and checkin
+
+Fleet workflow:
+
+- an agent enroll to fleet using an enrollmentAPiKey
+- Every n seconds agent is polling the checkin API to send events and check for new configuration
+
+### Agent enrollment
+
+An agent must enroll using the REST Api provided by fleet.
+When an agent enroll Fleet:
+
+- verify the API Key is a valid ES API key
+- retrieve the Saved Object (SO) associated to this api key id (this SO contains the configuration|policy id)
+- create an ES ApiKey unique to the agent for accessing kibana during checkin
```

Comment:
Where are the credentials stored that allow the creation of the ES API Keys that allow a check-in and the data output? Are these on the enrollment saved-object?

Reply:
The credentials are stored in a saved object named outputs, I am going to document this part in another PR